main.py

```diff
@@ -30,13 +31,14 @@ def create_account(self, account_type: str = "Normal", initial_balance: float =
         else:
             raise ValueError("Invalid account type.")
 
-        customer = Customer(account_number, account)
+        return account
+
+    def add_customer(self, customer: Customer):
         self.customers.append(customer)
-        return customer
 
-    def authenticate_customer(self, account_number: str, pin: str) -> Union[str, None]:
+    def authenticate_customer(self, pin: str) -> Union[str, None]:
```

Comment:
Now you can log into someones account just by providing a random pin number. What if 2 accounts will have the same pin numbers?